xwiki-platform-core/xwiki-platform-websocket/src/main/java/org/xwiki/websocket/script/WebSocketScriptService.java

```diff
@@ -84,9 +84,10 @@ public String url(String pathOrRoleHint)
             XWikiContext xcontext = this.xcontextProvider.get();
 
             StringBuilder path = new StringBuilder("/");
-            path.append(xcontext.getWiki().getWebAppPath(xcontext));
+            if (!xcontext.getWiki().getWebAppPath(xcontext).equals('/')) {
```

Comment:
At first look `getWebAppPath` cannot be null, but to be safer I'd use `!"/".equals(...)`.

Reply:
Sure. I'll change that.
The current code breaks if the value of `getWebAppPath` would ever become null, doesn't it?

Reply:
Sure. I'll change that.
The current code breaks if the value of `getWebAppPath` would ever become null, doesn't it?

Reply:
Yes right now if `getWebAppPath` returns null you'd always get a NullPointerException.